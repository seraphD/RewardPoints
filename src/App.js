import './App.css';
import { useEffect, useState } from 'react';
import { getPoints } from './api/points';

function App() {
  const [reward, setReward] = useState(0);
  const [points, setPoints] = useState(0);

  useEffect(() => {
    let newReward = 0;

    if (points > 50) {
      newReward += Math.min(50, points - 50);
    }

    if (points > 100) {
      newReward += 2 * (points - 100); 
    }

    setReward(newReward);
  }, [points]);

  const getNewPoints = () => {
    getPoints().then(p => setPoints(p))
  }

  return (
    <div className="App">
      <h1 data-testid="points">Current Points: {points}</h1>
      <h1 data-testid="reward">Reward: {reward}</h1>
      <button data-testid="button" onClick={getNewPoints}>get new points</button>
    </div>
  );
}

export default App;
