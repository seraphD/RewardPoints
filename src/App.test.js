import { render, fireEvent, screen, cleanup  } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import App from './App';

describe("Reward", () => {
  afterEach(cleanup);

  // snapshot test
  it('should match snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  // default
  it('should render default Reward App', () => {
    const wrapper = render(<App />);
    const pointEle = wrapper.getByTestId("points");
    const rewardEle = wrapper.getByTestId("reward");

    expect(pointEle).toHaveTextContent("0");
    expect(rewardEle).toHaveTextContent("0");
  });

  // click on button
  it('button should work', () => {
    const wrapper = render(<App />);
    const button = wrapper.getByTestId("points");
    button.onclick = jest.fn();

    expect(button.onclick).toHaveBeenCalledTimes(0);
    fireEvent.click(button);
    expect(button.onclick).toHaveBeenCalledTimes(1);
  })
})