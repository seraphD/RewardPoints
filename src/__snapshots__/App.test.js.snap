// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reward should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1
      data-testid="points"
    >
      Current Points: 0
    </h1>
    <h1
      data-testid="reward"
    >
      Reward: 0
    </h1>
    <button
      data-testid="button"
    >
      get new points
    </button>
  </div>
</DocumentFragment>
`;
